### PR TITLE
[MM-45729] Update playbook editor header UI

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR updates playbook editor header UI

https://user-images.githubusercontent.com/2340127/180268465-66a77c83-a3e0-4871-8f50-077f8343f15e.mp4

Bit confused about the `globe` icon in the [Figma](https://www.figma.com/file/f02HsYE1ftPoB44iWYoyIH/⬅%EF%B8%8F-LHS-Navigation?node-id=513%3A168245). Not sure what should happen when playbook is private. Another thing is that it looks kinda awkward next to the star, Star is clickable and I assume globe is clickable too and it does nothing... So I'm removing it for now. Let me know what you think

#### Ticket Link
  Fixes: https://mattermost.atlassian.net/browse/MM-45729

#### Checklist

- [ ] Telemetry updated
- [ ] Gated by experimental feature flag
- [ ] Unit tests updated

### CODE REVIEW

File: webapp/src/components/backstage/playbook_editor/playbook_editor.tsx [hunk 1]
```diff
@@ -174,48 +169,15 @@ const PlaybookEditor = () => {
...
-                                <span data-testid={'playbook-editor-title'}>
-                                    {playbook.title}
-                                </span>
-                            </Controls.TitleMenu>
-                            {privateTooltip}
-                            {archivedTooltip}
-                        </Heading>
-                    )}
-                </TextEdit>
+            <Header>
```
Comment (crspeller) [inline webapp/src/components/backstage/playbook_editor/playbook_editor.tsx:172]: I am a bit confused. It appears we have removed the ability to edit the header here?

**Review by crspeller — CHANGES_REQUESTED** [review]
Also noticed that the new version looks a little squished and removes the fancy scroll behavior. Is that intentional?

File: webapp/src/components/backstage/playbook_editor/playbook_editor.tsx (hunk 1 above)
Comment (iomodo) [reply webapp/src/components/backstage/playbook_editor/playbook_editor.tsx:172]: Yes, it's the new designs [here](https://www.figma.com/file/f02HsYE1ftPoB44iWYoyIH/⬅%EF%B8%8F-LHS-Navigation?node-id=422%3A150446). You can still edit it from the top left corner

**Review by calebroseland — APPROVED** [review]
Approving to not block, but I agree removing the ability to rename the playbook name/title in the larger one is a bit of a step back IMHO. 

If we're removing the header-title menu, I'd like to maybe see a pencil icon next to it?
CleanShot 2022-07-22 at 16 37 20@2x
CleanShot 2022-07-22 at 16 40 01@2x

**Review by calebroseland — COMMENTED** [review]
Also: 

Definitely in the scope of polish, but at some point this should probably be made more square.

CleanShot 2022-07-22 at 16 44 35@2x

**Review by calebroseland — COMMENTED** [review]
> Bit confused about the globe icon in the [Figma](https://www.figma.com/file/f02HsYE1ftPoB44iWYoyIH/%E2%AC%85%EF%B8%8F-LHS-Navigation?node-id=513%3A168245). Not sure what should happen when playbook is private. Another thing is that it looks kinda awkward next to the star, Star is clickable and I assume globe is clickable too and it does nothing... So I'm removing it for now. Let me know what you think



My $0.02: Channels relies on the LHS globe icon. Perhaps we could add a globe to the playbook icon to make it more obvious without adding extra UI in the playbook editor title area?


CleanShot 2022-07-22 at 16 49 47@2x
CleanShot 2022-07-22 at 16 50 15@2x

**Pull request closed without merging**